src/features/AddPaginationBar/ui/PaginationBar.tsx

```diff
@@ -0,0 +1,81 @@
+import { FC, useEffect, useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import { getPageCount, getSomePagination } from 'shared/lib/changeData';
+import { Paths } from 'shared/types';
+import './PaginationBar.css';
+
+interface PaginationProps {
+  totalResults: string | undefined;
+  activePage: number;
+}
+
+export const PaginationBar: FC<PaginationProps> = ({
+  totalResults,
+  activePage,
+}) => {
+  const offsetPagination = 370;
+  const limitMoviesOnPage = 10;
+  const [shift, addShist] = useState<number>(0);
+  const [countClick, setClick] = useState<number>(0);
+  const countPage: string[] = getPageCount(totalResults);
+
+  const countStart: number =
+    getSomePagination(countPage, limitMoviesOnPage, activePage) *
+    offsetPagination;
+
+  useEffect(() => {
+    addShist(countStart);
+  }, [countStart]);
+
+  const onClickPrev = () => {
+    addShist(shift - offsetPagination);
+    setClick(countClick - 1);
+  };
```

Comment:
```suggestion
  const onClickPrev = () => {
    addShist(shift - offsetPagination);
    setClick((prev) => prev - 1);
  };
```
в твоём варианте можно баг словить, если у countClick много раз будет менять значение